src/main/java/io/cdap/plugin/scd2/SCD2Plugin.java

```diff
@@ -181,6 +187,10 @@ public String getEndDateField() {
       return endDateField;
     }
 
+    public long getEndDateOffset() {
+      return endDateOffset == null ? 1L : endDateOffset;
```

Comment:
Do we need to guard against someone pass in a negative number?

Reply:
I added a check in validation to guard against a negative number

Reply:
Is it OK to have endDateOffset=0?

Reply:
Yea 0 is valid, it is ok to have a end time which is equal to next start time